tests/bluetooth/tester/src/btp_gatt.c

```diff
@@ -2290,6 +2290,39 @@ static uint8_t get_attr_val_rp(const struct bt_gatt_attr *attr, uint16_t handle,
 		net_buf_simple_add(buf, read);
 	} while (read == to_read);
 
+	/* use userdata only for tester own attributes */
+	if (IS_ARRAY_ELEMENT(server_db, attr)) {
+		const struct gatt_value *value = attr->user_data;
+
+		if ((rp->att_response == BT_ATT_ERR_SUCCESS) && (value->enc_key_size > 0)) {
```

Comment:
Minor optimization suggestion 
```suggestion
	if ((rp->att_response == BT_ATT_ERR_SUCCESS) && IS_ARRAY_ELEMENT(server_db, attr)) {
		const struct gatt_value *value = attr->user_data;

		if (value->enc_key_size > 0) {
```